src/cclib/parser/molproparser.py

```diff
@@ -916,6 +916,29 @@ def extract(self, inputfile, line):
                 self.atomcharges = {}
             self.atomcharges['mulliken'] = charges
 
+        if 'GRADIENT FOR STATE' in line:
+            reader = csv.reader(inputfile, delimiter=' ',
+                                skipinitialspace=True)
+            for _ in range(3):
+                next(reader)
+
+            def convert(x):
+                return (utils.convertor(x, 'hartree', 'eV')
+                        / utils.convertor(1, 'bohr', 'Angstrom'))
+
+            grad = []
+            lines_read = 0
+            while lines_read < self.natom:
+                line = next(reader)
+                if line:
+                    grad.append([convert(float(x)) for x in line[1:]])
+                    lines_read += 1
+            try:
+                self.grads.append(grad)
+            except AttributeError:
+                self.grads = [grad]
```

Comment:
We usually try to avoid try/except when setting attributes, and prefer

```python
if not has_attribute(self, 'grads'):
    self.grads = []
self.grads.append(grad)
```

Reply:
Changed accordingly